src/trajectory/Interpolated.cpp

```diff
@@ -100,6 +100,17 @@ void Interpolated::evaluateDerivative(double _t, int _derivative,
     _tangentVector.setZero();
   }
 
+  if(_t==mWaypoints[0].t)
```

Comment:
We're testing two floating point numbers of equality here. This is usually a bug, because the numbers may differ due to numerical precision issues. Please either add a tolerance or add a comment about why exact equality is appropriate here (e.g. because `t` was copied out of `mWaypoints[0].t`).